avr/cores/megacommand/MCL/ExtSeqTrack.cpp

```diff
@@ -44,27 +105,38 @@ void ExtSeqTrack::seq() {
   if ((MidiUart2.uart_block == 0) && (mute_until_start == false) &&
       (mute_state == SEQ_MUTE_OFF)) {
 
+    // the range we're interested in:
+    // [current timing bucket, micro >= timing_mid ... next timing bucket, micro
+    // < timing_mid]
+
+    uint16_t ev_idx, ev_end;
+
+    // Locate CURRENT
+    locate(step_count, ev_idx, ev_end);
+
+    // Go over CURRENT
+    for (; ev_idx != ev_end; ++ev_idx) {
+      auto u = events[ev_idx].micro_timing;
+      if (u >= timing_mid && u - timing_mid == mod12_counter) {
+        handle_event(ev_idx);
```

Comment:
fyi: when lock events are added, they need to be sent before note_on/off .
This ensures machine params are updated before sound trigger.

Reply:
ah, yes.. need to do it in the editor part

Reply:
also, velocity locks should be treated specially because that's part of the midi note event.